# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpBackgroundChannelChild.h
# Commit: b8f55064bfeb
# Full Hash: b8f55064bfeb151f77c084d04f3def3bcd96679a
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-30 10:00:42
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P1 allow OnStartRequest go thru pBg IPC, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73529
# ==============================================================================

diff -r b5ac0f98346f -r b8f55064bfeb netwerk/protocol/http/HttpBackgroundChannelChild.h
--- a/netwerk/protocol/http/HttpBackgroundChannelChild.h	Tue Jun 30 00:31:12 2020 +0000
+++ b/netwerk/protocol/http/HttpBackgroundChannelChild.h	Tue Jun 30 00:32:25 2020 +0000
@@ -51,6 +51,11 @@
 #endif
 
  protected:
+  IPCResult RecvOnStartRequest(const nsHttpResponseHead& aResponseHead,
+                               const bool& aUseResponseHead,
+                               const nsHttpHeaderArray& aRequestHeaders,
+                               const HttpChannelOnStartRequestArgs& aArgs);
+
   IPCResult RecvOnTransportAndData(const nsresult& aChannelStatus,
                                    const nsresult& aTransportStatus,
                                    const uint64_t& aOffset,